### ISSUE TITLE
region errors when managing source and destination for AWS S3 cross region replication in module.

### ISSUE THREAD

**Issue by grimm26** [hashicorp/terraform#12287]
Using terraform 0.8.7 with terragrunt 0.10.3

I [wrote a module](https://www.dropbox.com/s/qnr6capfg4nq047/s3_module.zip?dl=0) to create AWS S3 buckets with corresponding cross region replication buckets.  The debug plan output is included there as well.

I added the lifecycle_rule section to both the original and destination bucket in the module and when trying to plan that for existing buckets where the original is in us-east-2 and the replication destination is in us-east-1 and I have AWS_DEFAULT_REGION=us-east-2:
```
....
Error refreshing state: 1 error(s) occurred:

* aws_s3_bucket.b: error reading S3 bucket "enova-mktest-standard.us-east-1": BucketRegionError: incorrect region, the bucket is not in 'us-east-2' region
        status code: 301, request id:
exit status 1
```

Is there some other strategy that I should be using in managing a bucket's accompanying replication destination?

**Comment by grimm26** [comment]
Nevermind, I got my state all hosed up :)

**Comment by chunyong-lin** [comment]
Hi @grimm26 How do you solve this problem? I ran into same issue last night. 

I got error when I do `terraform plan`
```
Error refreshing state: 1 error(s) occurred:

... error reading S3 bucket 
... BucketRegionError: incorrect region, the bucket is not in 'us-east-1' region
	status code: 301, request id:
```

I wanted to set the region/provider of this S3 bucket to ap-southeast-1. Right now, i can neither delete nor change this S3 bucket. Stuck. I am new to terraform, any suggestion would be helpful for me. Thanks!

**Comment by grimm26** [comment]
I believe I ended up removing the state entries for the buckets and re-importing them.  I had been incrementally doing experimental stuff as I developed and I think I just got things in a jumbled state.

Most of the time when I hit an issue like this it is because terraform has no refernece as to what region the resource should be in because I either removed it from the tf file to delete it (destroy first then remove from tf file is more reliable in my experience) or I am adding importing something out of the blue as a resource in a module that uses multiple aws providers for different regions.  

You also need to make sure that all relevant resources have an explicit provider set in them when you are working with multiple regions together.  I hope that helps.
